language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/signature/SignatureHelpUtil.java

```diff
@@ -344,12 +347,24 @@ private static boolean withinTextRange(int position, NonTerminalNode node) {
         }
 
         if (tokenAtCursor.get().kind() == SyntaxKind.FUNCTION_CALL) {
-            String funcName = ((SimpleNameReferenceNode) ((FunctionCallExpressionNode) tokenAtCursor.get())
-                    .functionName()).name().text();
-            List<Symbol> visibleSymbols = context.visibleSymbols(context.getPosition());
-            return visibleSymbols.stream().filter(symbol -> symbol.kind() == FUNCTION && symbol.name().equals(funcName))
-                    .map(symbol -> (FunctionSymbol) symbol)
-                    .findAny();
+            NameReferenceNode nameReferenceNode = ((FunctionCallExpressionNode) tokenAtCursor.get()).functionName();
+            String funcName;
+            Predicate<Symbol> symbolPredicate = symbol -> symbol.kind() == FUNCTION;
+            List<Symbol> filteredContent;
+            if (nameReferenceNode.kind() == SyntaxKind.QUALIFIED_NAME_REFERENCE) {
+                funcName = ((QualifiedNameReferenceNode) nameReferenceNode).identifier().text();
+                filteredContent = QNameReferenceUtil.getModuleContent(context,
+                        (QualifiedNameReferenceNode) nameReferenceNode,
+                        symbolPredicate.and(symbol -> symbol.name().equals(funcName)));
+            } else {
+                funcName = ((SimpleNameReferenceNode) nameReferenceNode).name().text();
+                List<Symbol> visibleSymbols = context.visibleSymbols(context.getCursorPosition());
+                filteredContent = visibleSymbols.stream()
+                        .filter(symbolPredicate.and(symbol -> symbol.name().equals(funcName)))
+                        .collect(Collectors.toList());
```

Comment:
Do we have to take the visible symbols and filter? Can't we use `symbol()` for this?

Reply:
This particular operation is triggered frequently for the incomplete sources and sometimes the semantic API might not capture the correct symbol because of the resilient parsing. Hence we have to use this particular approach.

Reply:
Ack.